mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -255,18 +256,37 @@ func (sp *StateProcessor[
 		)
 	}
 
+	// Ensure the EVM inflation withdrawal is the first withdrawal.
+	if numWithdrawals == 0 {
+		return ErrNoWithdrawals
+	}
+	if !expectedWithdrawals[0].GetAddress().Equals(
+		sp.cs.EVMInflationAddress(),
+	) ||
+		expectedWithdrawals[0].GetAmount() != math.Gwei(
+			sp.cs.EVMInflationPerBlock(),
+		) {
+		return ErrFirstWithdrawalNotEVMInflation
+	}
+
 	// Compare and process each withdrawal.
 	for i, wd := range expectedWithdrawals {
 		// Ensure the withdrawals match the local state.
 		if !wd.Equals(payloadWithdrawals[i]) {
 			return errors.Wrapf(
-				ErrNumWithdrawalsMismatch,
-				"withdrawals do not match expected %s, got %s",
-				spew.Sdump(wd), spew.Sdump(payloadWithdrawals[i]),
+				ErrWithdrawalMismatch,
```

Comment:
is the range above correct? I would expect the indexes to be shifted all by one. Something like:
`for i:=1; i <expectedWithdrawals; i++`
In this way we also simplify the loop, since we don't need to test for i==0

Reply:
would agree that the validation should just be for all, as we don't need special check for i==0.